src/plugins/ds_json.c

```diff
@@ -150,10 +150,16 @@ srpds_json_store_(const char *path, const struct lyd_node *mod_data, const char
                 strerror(errno));
     }
 
-    ly_out_free(out, NULL, 1);
-    if (fd > -1) {
-        close(fd);
+    if (out) {
+        /* will call fclose(), so close() should not be called again */
+        ly_out_free(out, NULL, 1);
+    } else if (fd > -1) {
+        if (close(fd) < 0) {
+            srplg_log_errinfo(&err_info, srpds_name, NULL, SR_ERR_SYS, "Failed to close fd for \"%s\" (%s).", path,
+                    strerror(errno));
+        }
     }
+
```

Comment:
I do not think the error check is needed. I would make only one small change, use `0` instead of `1` as the last parameter of `ly_out_free()`.

Reply:
But in that case, doesn't `fclose(fp)` need to be called, to avoid leaking any resources associated with `FILE*` ?

Reply:
Yeah, once again proving the github diff is often missing important context. Fixed myself.